packages/firebase_vertexai/firebase_vertexai/lib/src/imagen_api.dart

```diff
@@ -0,0 +1,214 @@
+      _ => throw FormatException(
+          'Unhandled ImagenSafetyFilterLevel format', jsonObject),
+    };
+  }
+
+  @override
+  String toString() => name;
+}
+
+/// Allow generation of people by the model.
+/// If not specified, the default value is "allow_adult".
+enum ImagenPersonFilterLevel {
+  /// Disallow the inclusion of people or faces in images.
+  blockAll('dont_allow'),
+
+  /// Allow generation of adults only.
+  allowAdult('allow_adult'),
+
+  /// Allow generation of people of all ages.
+  allowAll('allow_all');
+
+  const ImagenPersonFilterLevel(this._jsonString);
+
+  final String _jsonString;
+
+  /// Convert to json format
+  String toJson() => _jsonString;
+
+  // ignore: unused_element
+  static ImagenPersonFilterLevel _parseValue(Object jsonObject) {
+    return switch (jsonObject) {
+      'dont_allow' => ImagenPersonFilterLevel.blockAll,
+      'allow_adult' => ImagenPersonFilterLevel.allowAdult,
+      'allow_all' => ImagenPersonFilterLevel.allowAll,
+      _ => throw FormatException(
+          'Unhandled ImagenPersonFilterLevel format', jsonObject),
+    };
+  }
+
+  @override
+  String toString() => name;
+}
+
+/// A class representing safety settings for image generation.
+///
+/// It includes a safety filter level and a person filter level.
+final class ImagenSafetySettings {
+  /// Constructor
+  ImagenSafetySettings(this.safetyFilterLevel, this.personFilterLevel);
+
+  /// The safety filter level
+  final ImagenSafetyFilterLevel? safetyFilterLevel;
+
+  /// The person filter level
+  final ImagenPersonFilterLevel? personFilterLevel;
+
+  /// Convert to json format.
+  Object toJson() => {
+        if (safetyFilterLevel != null)
+          'safetySetting': safetyFilterLevel!.toJson(),
+        if (personFilterLevel != null)
+          'personGeneration': personFilterLevel!.toJson(),
+      };
+}
+
+/// The aspect ratio for the image. The default value is "1:1".
+enum ImagenAspectRatio {
+  /// Square (1:1).
+  square1x1('1:1'),
+
+  /// Portrait (9:16).
+  portrait9x16('9:16'),
+
+  /// Landscape (16:9).
+  landscape16x9('16:9'),
+
+  /// Portrait (3:4).
+  portrait3x4('3:4'),
+
+  /// Landscape (4:3).
+  landscape4x3('4:3');
+
+  const ImagenAspectRatio(this._jsonString);
+
+  final String _jsonString;
+
+  /// Convert to json format
+  String toJson() => _jsonString;
+
+  // ignore: unused_element
+  static ImagenAspectRatio _parseValue(Object jsonObject) {
+    return switch (jsonObject) {
+      '1:1' => ImagenAspectRatio.square1x1,
+      '9:16' => ImagenAspectRatio.portrait9x16,
+      '16:9' => ImagenAspectRatio.landscape16x9,
+      '3:4' => ImagenAspectRatio.portrait3x4,
+      '4:3' => ImagenAspectRatio.landscape4x3,
+      _ =>
+        throw FormatException('Unhandled ImagenAspectRatio format', jsonObject),
+    };
+  }
+
+  @override
+  String toString() => name;
+}
+
+/// Configuration options for image generation.
+final class ImagenGenerationConfig {
+  /// Constructor
```

Comment:
[nit] Drop this doc since it isn't saying anything that isn't clear from the signature.

https://dart.dev/effective-dart/documentation#avoid-redundancy-with-the-surrounding-context